pkg/workload/workload.go

```diff
@@ -244,6 +245,25 @@ func UpdateStatusIfChanged(ctx context.Context,
 	return UpdateStatus(ctx, c, wl, conditionType, conditionStatus, reason, message, managerPrefix)
 }
 
+func UpdateAdmittedCondition(
+	ctx context.Context,
+	c client.Client,
+	wl *kueue.Workload,
+	conditionStatus metav1.ConditionStatus,
+	reason, message string) error {
```

Comment:
This only really works to make the Admitted condition to false (because it would set the `.admission` field to `nil`). So a better name would be:

```suggestion
func UnsetAdmissionWithCondition(
	ctx context.Context,
	c client.Client,
	wl *kueue.Workload,
	reason, message string) error {
```

Reply:
done